HSP2tools/readWDM.py

```diff
@@ -149,137 +171,199 @@ def readWDM(wdmfile, hdffile, jupyterlab=True):
+
+    #@njit 
+    @staticmethod
+    def _inttostr(i):
+        return chr(i & 0xFF) + chr(i>>8 & 0xFF) + chr(i>>16 & 0xFF) + chr(i>>24 & 0xFF)
+
+    #@njit 
+    @staticmethod
+    def bits_to_date(x):
+        year = x >> 26
+        month = x >> 22 & 0xf
+        day = x >> 17 & 0x1f
+        hour = x >> 12 & 0x1f
+        minute = x >> 6 & 0x3f
+        second = x & 0x3f
+        return year, month, day, hour, minute, second
+
+    #@njit 
+    @staticmethod
+    def date_to_bits(year, month, day, hour, minute, second):
+        x = year << 26 | month << 22 | day << 17 | hour << 12 | minute << 6 | second 
+        return x
+
+    #@njit 
+    @staticmethod
+    def increment_date(date, timecode, timestep):
+        year, month, day, hour, minute, second = WDMReader.bits_to_date(date)
+        
+        if timecode == 7: year += 100 * timestep
+        elif timecode == 6 : year += timestep
+        elif timecode == 5 : month += timestep
+        elif timecode == 4 : day += timestep
+        elif timecode == 3 : hour += timestep
+        elif timecode == 2 : minute += timestep
+        elif timecode == 1 : second += timestep
+
+        return WDMReader.correct_date(year, month, day, hour, minute, second)
+
+    #@njit 
+    @staticmethod
+    def correct_date(year, month, day, hour, minute, second):
+        while second >= 60:
+            second -= 60
+            minute += 1
+        while minute >= 60:
+            minute -= 60
+            hour += 1
+        while hour >= 24:
+            hour -= 24
+            day += 1
+        while day > WDMReader._days_in_month(year, month):
+            day -= WDMReader._days_in_month(year, month)
+            month += 1
+        while month > 12:
+            month -= 12
+            year += 1
+        return WDMReader.date_to_bits(year, month, day, hour, minute, second)
+        
+    #@njit 
+    @staticmethod
+    def _days_in_month(year, month):
+        if month > 12: month %= 12
+        
+        if month in (1,3,5,7,8,10,12):
+            return 31
+        elif month in (4,6,9,11):
+            return 30
+        elif month == 2:
+            if WDMReader._is_leapyear(year): return 29
+            else: return 28
+
+    #@njit 
+    @staticmethod
+    def _is_leapyear(year):
+        if year % 400 == 0:
+            return True
+        if year % 100 == 0:
+            return False
+        if year % 4 == 0:
+            return True
         else:
-            nval = -1  # dont know how to convert
-
-    nval = int(nval)
-    if nval == -1:  # conversion problem
-        print('Conversion problem (tcode ', str(tcode), ', ', str(ltcode), '), (tstep ', str(tstep), ',', str(ltstep), '), (comp ', str(comp), ')')
-    else:
-        print('Conversion complete (tcode ', str(tcode), ', ', str(ltcode), '), (tstep ', str(tstep), ',', str(ltstep), '), (nval ',
-              str(nval) + ',', str(lnval), ')')
-
-    return nval
+            return False
+    
+    @njit
```

Comment:
When I resolved the merge conflicts in commit [d64c25a](https://github.com/LimnoTech/HSPsquared/commit/d64c25a58c5f478c7bdf96f8932ecada5d4f810a), I failed to make this a class method. As this is currently written, it will produce a runtime error. Fix by either adding the staticmethod decorator or a self argument.